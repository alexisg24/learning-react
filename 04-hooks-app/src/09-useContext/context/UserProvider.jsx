import { useState } from 'react'
import { UserContext } from './UserContext'

const UserProvider = ({ children }) => {
  const [user, setUser] = useState()
  const newUser = () => {
    setUser({ id: 1, name: 'User', email: 'user@example.com' })
  }
  return (
    <UserContext.Provider value={{ user, newUser }}>
      {children}
    </UserContext.Provider>
  )
}

export { UserProvider }
