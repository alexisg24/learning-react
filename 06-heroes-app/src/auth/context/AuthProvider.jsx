import { useAuth } from '../hooks/useAuth'
import { AuthContext } from './AuthContext'
const AuthProvider = ({ children }) => {
  const { state, login, logout } = useAuth()
  return (
    <AuthContext.Provider value={{ login, logout, ...state }}>
      {children}
    </AuthContext.Provider>
  )
}

export { AuthProvider }
