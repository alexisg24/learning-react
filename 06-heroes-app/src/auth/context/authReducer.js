import { types } from '../types/types'

export const authReducer = (state = {}, action) => {
  const ACTIONS = {
    [types.LOGIN]: { ...state, logged: true, user: action?.payload },
    [types.LOGOUT]: { ...state, logged: false, user: null }
  }

  return ACTIONS[action?.type] ?? state
}
