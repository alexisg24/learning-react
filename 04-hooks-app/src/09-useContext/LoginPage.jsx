import { useContext } from 'react'
import { UserContext } from './context/UserContext'

const LoginPage = () => {
  const { user, newUser } = useContext(UserContext)
  return (
    <>
      <h1>LoginPage</h1>
      <hr />
      <pre aria-label='user-info'>
        {JSON.stringify(user, null, 3)}
      </pre>
      <button
        className='btn btn-primary'
        aria-label='newUser-button'
        onClick={() => newUser()}
      >Set User
      </button>
    </>
  )
}

export { LoginPage }
