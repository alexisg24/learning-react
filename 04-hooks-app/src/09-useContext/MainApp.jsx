import { AppMenu } from './components/AppMenu'
import { RoutesPath } from './components/RoutesPath'
import { UserProvider } from './context/UserProvider'

const MainApp = () => {
  return (
    <>
      <AppMenu />
      <hr />
      <UserProvider>
        <RoutesPath />
      </UserProvider>
    </>
  )
}

export { MainApp }
