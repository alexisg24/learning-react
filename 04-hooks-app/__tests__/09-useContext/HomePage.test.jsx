import { render, screen } from '@testing-library/react'
import { HomePage } from '../../src/09-useContext/HomePage'
import { UserContext } from '../../src/09-useContext/context/UserContext'

describe('Tests in <HomePage />', () => {
  const user = { id: 1, name: 'Alexis' }

  test('should render the component without the user', () => {
    const { container } = render(
      <UserContext.Provider value={{ user: null }}>
        <HomePage />
      </UserContext.Provider>
    )
    expect(container).toMatchSnapshot()

    const preTag = screen.getByLabelText('pre-selector')
    expect(preTag.innerHTML).toBe('null')
  })

  test('should render the component with the user', () => {
    render(
      <UserContext.Provider value={{ user }}>
        <HomePage />
      </UserContext.Provider>
    )
    const h1Tag = screen.getByRole('heading', { level: 1 })
    expect(h1Tag.innerHTML).toContain(user.name)
    const preTag = screen.getByLabelText('pre-selector')
    expect(preTag.innerHTML).toBe(JSON.stringify(user, null, 3))
  })
})
