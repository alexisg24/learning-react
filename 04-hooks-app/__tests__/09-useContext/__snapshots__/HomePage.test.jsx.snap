// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests in <HomePage /> should render the component without the user 1`] = `
<div>
  <h1>
    HomePage 
    <small />
  </h1>
  <hr />
  <pre
    aria-label="pre-selector"
  >
    null
  </pre>
</div>
`;
